go/filer/redis_store/directory_manager_test.go

```diff
@@ -0,0 +1,264 @@
+	// /a/b/f should exist
+	dir = to
+	did, err = dm.FindDirectory(dir)
+	if did != 4 {
+		t.Errorf("new dir %s has a wrong dir id:%d, 4 is expected", dir, did)
+	}
+	// /a/b/f/e also should exist
+	dir = to + "/e"
+	did, err = dm.FindDirectory(dir)
+	if did != 6 {
+		t.Errorf("new dir %s has a wrong dir id:%d, 6 is expected", dir, did)
+	}
+
+	/*
+	 * list a dir's sub-directories
+	 *
+	 */
+	dir = "/a/b"
+	entries, err := dm.ListDirectories(dir)
+	if !(entries[0].Name == "d" && entries[1].Name == "f") {
+		t.Errorf("get entries:%v, expect 'd', 'f'", entries)
+	}
+	/*
+		* delete a dir: /a/b/f/e
+		 *
+	*/
+	dir = "/a/b/f"
+	err = dm.DeleteDirectory(dir)
+	if err != nil {
+		t.Errorf("delete dir:%s error:%v", dir, err)
+	}
+	// now the deleted dir should not exist
+	did, err = dm.FindDirectory(dir)
+	if err != nil {
+		t.Errorf("get deleted dir %s error:%v", dir, err)
+	}
+	if did != 0 {
+		t.Errorf("the dir %s is not deleted!", dir)
+	}
+	// now the deleted dir's sub-directory also should not exist
+	dir = "/a/b/f/e"
+	did, err = dm.FindDirectory(dir)
+	if err != nil {
+		t.Errorf("get deleted dir %s error:%v", dir, err)
+	}
+	if did != 0 {
+		t.Errorf("the dir %s is not deleted!", dir)
+	}
+}
+
+func TestFiles(t *testing.T) {
+	// put one file
+	fname := "/a/b/c/test.txt"
+	fid := "1,23"
+	err := dm.PutFile(fname, fid)
+	if err != nil {
+		t.Errorf("put file %s error:%v", fname, err)
+	}
+	// put another file
+	fname = "/a/b/c/abc.txt"
+	fid = "1,234"
+	err = dm.PutFile(fname, fid)
+	if err != nil {
+		t.Errorf("put file %s error:%v", fname, err)
+	}
+	//get file
+	id, err := dm.FindFile(fname)
+	if err != nil {
+		t.Errorf("get file %s error:%v", fname, err)
+	}
+	if id != fid {
+		t.Errorf("get wrong fid %s, expect %s", id, fid)
+	}
+	//list files
+	lastFileName := "abc.txt"
+	dir := filepath.Dir(fname)
+	files, err := dm.ListFiles(dir, lastFileName, 10)
+	if err != nil {
+		t.Errorf("list files for %s error:%v", dir, err)
+	}
+	if files[0].Name != "test.txt" {
+		t.Errorf("files list order wrong, first file is %s, expect %s", files[0].Name, "test.txt")
+	}
+	//delete file
+	id, err = dm.DeleteFile(fname)
+	if err != nil {
+		t.Errorf("delete file get error:%v", err)
+	}
+	if id != fid {
+		t.Errorf("delete file return wrong fid %s, expect %s", id, fid)
+	}
+}
+
+func clearRedisKeys(client *redis.Client, dirKeyPrefix string, dirMaxIdKey string) error {
+	result, err := client.Keys(dirKeyPrefix + "*").Result()
+	if err != nil {
+		fmt.Println("get redis keys error:", err)
+		return err
+	}
+	if len(result) > 0 {
+		n, err := client.Del(result...).Result()
+		if err != nil {
+			fmt.Println("del keys error:", err)
+		} else {
+			fmt.Println("del", n, " keys.")
+		}
+	}
+	n, err := client.Del(dirMaxIdKey).Result()
+	if err != nil {
+		fmt.Printf("del dirMaxIdKey:%s, error:%v\n", dirMaxIdKey, err)
+	} else {
+		fmt.Println("del", n, " keys.")
+	}
+	return err
+}
+func TestMain(m *testing.M) {
+	redisClient := redis.NewTCPClient(&redis.Options{
```

Comment:
Having a command line for the test would also help. With it, we can just run the test pointing to any redis server, instead of requiring users to install redis just to run the test.